### ISSUE TITLE
Vault UI not working for Allowed extensions for login@github.com for SSH Certs when user submit a key to sign for Github 

### ISSUE THREAD

**Issue by DtwoRtwo** [hashicorp/vault#11444]
**Describe the bug**
I created a SSH key sign role to sign SSH certs using the Vault UI. While working with Vault client, we are able to sign SSH keys with Vault SSH CA over APIs. Vault client can include extensions such as "login@github.com":"git-example-username" for GitHub logon but when we do that same with the UI then this extension is shown null. 

**To Reproduce**
1. Open the UI
2. Define a role. Added "login@github.com" to Allowed extension
3. Use the role to sign a key. Include { "login@github.com":"hello"} in extension as part of signing request
4. inspect the cert with ssh-keygen.exe -Lf cert-name.pub
5. Output shows Extensions: (none)

**Expected behavior**
While using the Vault UI, if follow above process as we do with vault client then output should be following of ssh-keygen.exe -Lf cert-name.pub:
Under Extension should see:
login@github.com Unknown Option (len 20). Len is lengh of  input provided as part of extension when signing. 

**Environment:**
* Vault Server Version (retrieve with `vault status`):
* Vault CLI Version (retrieve with `vault version`):
* Server Operating System/Architecture:
Vault server UI running 1.7.1 
Vault server configuration file(s):

```hcl
# Paste your Vault config here.
# Be sure to scrub any sensitive values
```

**Additional context**

**Comment by DtwoRtwo** [comment]
It seems to related with Fix: sign SSH key with extensions on web UI #11194

**Comment by hazcod** [comment]
Does Vault check the value of `login@github.com` at all?

**Comment by mickael-hc** [comment]
Thanks for opening this issue @DtwoRtwo ! This appears to have been resolved by the PR you linked above, and was released in Vault 1.8.0. I’ll close the issue for now, but please don’t hesitate to comment/reopen if you are still experiencing issues in Vault 1.8.0 or above.

Quickly following up on the conversation above:
> Define a role. Added login@github.com to Allowed extension

> Does Vault check the value of login@github.com at all? This poses a security risk.

As alluded to by @hazcod above, keep in mind that adding `login@github.com` to [allowed_extensions will mean that anyone with the Vault role to sign SSH certificates will be able to create certificates with any value in that field](https://www.vaultproject.io/docs/secrets/ssh/signed-ssh-certificates#signing-key-role-configuration) (and therefore impersonate, via SSH) arbitrary users in the organization - Vault does not further validate the contents of the `login@github.com` extension contents. If you aren’t using a trusted service to request these certificates from Vault, you can consider:
* using a templated policy for the signing role that populates the `login@github.com` extension with values in Vault mapped to the user, and not adding `login@github.com` to allowed_extensions for this role 
* or using control groups (https://learn.hashicorp.com/tutorials/vault/control-groups) to set up an approval flow to have another user validate the contents of the GitHub-specific field

**Comment by radek-sprta** [comment]
>     * using a templated policy for the signing role that populates the `login@github.com` extension with values in Vault mapped to the user, and not adding `login@github.com` to allowed_extensions for this role
> 
I am curious as to what exactly have you meant by that. Identity templating is not supported for `allowed_parameters` in policies. So I assume using an external templating engine to pre-generate policies for all users?
